test/functional/lua/net_spec.lua

```diff
@@ -39,6 +39,37 @@ describe('vim.net.request', function()
     )
   end)
 
+  it('detects filetype for remote content', function()
+    t.skip(skip_integ, 'NVIM_TEST_INTEG not set: skipping network integration test')
+
+    local ft = exec_lua([[
+      vim.cmd('runtime! plugin/nvim/net.lua')
+      vim.cmd('runtime! filetype.lua')
+      -- github raw dump of a small lua file in the neovim repo
+      vim.cmd('edit https://raw.githubusercontent.com/neovim/neovim/master/runtime/syntax/tutor.lua')
+      vim.wait(2000, function() return vim.bo.filetype ~= '' end)
+      return vim.bo.filetype
+    ]])
+
+    assert(ft == 'lua', 'Expected filetype to be "lua", got: ' .. tostring(ft))
+  end)
+
+  it('removes the modified flag from the buffer for remote content', function()
```

Comment:
Can be done in 1 test. Duplicated in https://github.com/neovim/neovim/pull/36610/commits/b6cc7c625da1c4189664e2035312c83ba67c2112

Thank you for the fix!